pkg/pool-utils/contracts/oracle/PoolPriceOracle.sol

```diff
@@ -77,6 +77,24 @@ abstract contract PoolPriceOracle is IPoolPriceOracle, IPriceOracle {
         return Buffer.SIZE;
     }
 
+    /**
+     * @dev Manually fill oracle slots with dummy data, so that subsequent swaps filling them with real data
+     * cost less gas. Initialize by range, since filling a slot costs roughly 20k gas, so trying to fill the
+     * entire buffer at once would run out of gas.
+     */
+    function initializeOracle(uint256 startSlot, uint256 endSlot) external {
+        _require(startSlot < endSlot && endSlot <= Buffer.SIZE, Errors.OUT_OF_BOUNDS);
+
+        // Cannot use zero, or it will not actually do anything (uninitialized entries are already 0)
+        // Cannot use 0x01, since the timestamp is in the LSB position, and needs to be zero for invalid entries
+        bytes32 initSample = bytes32(0xba10000000000000000000000000000000000000000000000000000000000000);
```

Comment:
```suggestion
        // Uninitialized samples are identified by a zero timestamp -- all other fields are ignored, so any non-zero value
        // with a zero timestamp suffices.
        bytes32 initSample = Samples.pack(1, 0, 0, 0, 0, 0, 0);
```

Reply:
I don't like that constant being coupled to the layout of the samples - it's too error prone. We should also have tests here to prevent such issues.

Reply:
I added tests for it